subprojects/build-cache/src/test/groovy/org/gradle/caching/internal/origin/OriginMetadataFactoryTest.groovy

```diff
@@ -17,22 +17,17 @@
 package org.gradle.caching.internal.origin
 
 import org.gradle.caching.internal.CacheableEntity
-import org.gradle.internal.id.UniqueId
 import org.gradle.internal.remote.internal.inet.InetAddressFactory
-import org.gradle.internal.time.Clock
-import org.gradle.util.GradleVersion
 import spock.lang.Specification
 
 class OriginMetadataFactoryTest extends Specification {
     def entry = Mock(CacheableEntity)
-    def timeProvider = Mock(Clock)
     def inetAddressFactory = Mock(InetAddressFactory)
     def rootDir = Mock(File)
-    def buildInvocationId = UniqueId.generate()
-    def factory = new OriginMetadataFactory(timeProvider, inetAddressFactory, rootDir, "user", "os", GradleVersion.version("3.0"), buildInvocationId)
+    def buildInvocationId = UUID.randomUUID().toString()
```

Comment:
could we store the UUID instead of a String?
We should be able to deserialize it via its constructor.

Reply:
I wouldn't bother for a unit test. Or are you suggesting some other place where things would be changed?